src/components/layout/nav/MobileNav/MobileNav.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect } from 'react'
+
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+
+import { SwipeableDrawer, Hidden, Grid } from '@mui/material'
+import FavoriteIcon from '@mui/icons-material/Favorite'
+
+import LocaleButton from '../../LocaleButton'
+import DonationMenuMobile from '../DonationMenuMobile'
+import ProjectMenuMobile from '../ProjectMenuMobile'
+import { AuthLinks } from '../AuthLinks/AuthLinks'
+import { routes } from 'common/routes'
+
+import {
+  CloseButton,
+  NavMenuWrapper,
+  LocaleButtonWrapper,
+  StyledPodkrepiIcon,
+  DonateButton,
+} from './MobileNav.styled'
+
+type NavDeckProps = {
+  mobileOpen: boolean
+  setMobileOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+export default function MobileNav({ mobileOpen, setMobileOpen }: NavDeckProps) {
+  const { t } = useTranslation()
+
+  const router = useRouter()
+  const closeNavMenu = () => setMobileOpen(false)
+
+  // Register route change event handlers
+  useEffect(() => {
+    router.events.on('routeChangeStart', closeNavMenu)
+
+    return () => {
+      router.events.off('routeChangeStart', closeNavMenu)
+    }
+  }, [])
+
+  return (
+    <Hidden mdUp implementation="css">
+      <SwipeableDrawer
+        anchor="right"
+        open={mobileOpen}
+        variant="temporary"
+        ModalProps={{ keepMounted: true }}
+        onOpen={() => setMobileOpen(true)}
+        onClose={closeNavMenu}>
+        <CloseButton edge="end" fontSize="inherit" onClose={closeNavMenu} />
+        <NavMenuWrapper>
+          <Grid item textAlign="center">
+            <StyledPodkrepiIcon />
+          </Grid>
+          <Grid item>
+            <DonationMenuMobile />
+          </Grid>
+          <Grid item>
+            <ProjectMenuMobile />
+          </Grid>
+          <AuthLinks />
+          <LocaleButtonWrapper>
+            <LocaleButton />
+          </LocaleButtonWrapper>
```

Comment:
Here we mix Grid items with Box elements.

I'm not sure it will propagate the grid styles and spacing accordingly without another Grid item.